instance/models/openedx_instance.py

```diff
@@ -249,54 +243,107 @@ def delete(self, *args, **kwargs):
         self.deprovision_swift()
         super().delete(*args, **kwargs)
 
-    @property
-    def appserver_set(self):
-        """
-        Get the set of OpenEdxAppServers owned by this instance.
-        """
-        return self.ref.openedxappserver_set
-
-    def set_appserver_active(self, appserver_id):
+    def set_dns_records(self):
         """
-        Mark the AppServer with the given ID as the active one.
+        Create CNAME records for the domain names of this instance pointing to the load balancer.
         """
-        app_server = self.appserver_set.get(pk=appserver_id)  # Make sure the AppServer is owned by this instance
-        self.logger.info('Making %s active for instance %s...', app_server.name, self.name)
-        public_ip = app_server.server.public_ip
+        load_balancer_domain = self.load_balancing_server.domain.rstrip(".") + "."
 
         self.logger.info('Updating DNS: LMS at %s...', self.internal_lms_domain)
         lms_domain = tldextract(self.internal_lms_domain)
         gandi.set_dns_record(
             lms_domain.registered_domain,
-            type='A', name=lms_domain.subdomain, value=public_ip
+            type='CNAME', name=lms_domain.subdomain, value=load_balancer_domain
         )
 
         self.logger.info('Updating DNS: LMS preview at %s...', self.internal_lms_preview_domain)
         lms_preview_domain = tldextract(self.internal_lms_preview_domain)
         gandi.set_dns_record(
             lms_preview_domain.registered_domain,
-            type='CNAME', name=lms_preview_domain.subdomain, value=lms_domain.subdomain
+            type='CNAME', name=lms_preview_domain.subdomain, value=load_balancer_domain
         )
 
         self.logger.info('Updating DNS: Studio at %s...', self.internal_studio_domain)
         studio_domain = tldextract(self.internal_studio_domain)
         gandi.set_dns_record(
             studio_domain.registered_domain,
-            type='CNAME', name=studio_domain.subdomain, value=lms_domain.subdomain
+            type='CNAME', name=studio_domain.subdomain, value=load_balancer_domain
         )
 
+    def get_load_balancer_configuration(self):
+        """
+        Return the haproxy configuration fragment and backend map for this instance.
+        """
+        if self.active_appserver:
+            backend_name = "be-{}".format(self.active_appserver.server.name)
+            server_name = "appserver-{}".format(self.pk)
+            ip_address = self.active_appserver.server.public_ip
+        else:
+            backend_name = "be-preliminary-page-{}".format(self.pk)
+            server_name = "preliminary-page"
+            ip_address = settings.PRELIMINARY_PAGE_SERVER_IP
```

Comment:
The "please wait" page can only be served via HTTPS once a certificate has been retrieved, and a certificate can only be retrieved once the DNS entry points to the load balancer.

With the current setup, the following is happening: The links in the instance manager are still HTTP links (and I don't plan to change this).  Clicking on the link before DNS has propagated will result in the "please wait" page being served via HTTP.  Once the load balancer detects that DNS has propagated, it will request a cert and start serving the page via HTTPS (well, it doesn't actually "serve" it, but you know what I mean).  Since users and the load-balancer might see the results of DNS propagation at different times, and since the load balancer only checks every five minutes whether it needs to request new certs, there can be a gap between the two ways of serving the page.

If we change where the wildcard entry point to the load balancer instead, the load balancer will be able to request a certificate more or less immediately, and then be able to serve the "please wait" page without interruption, so yes, that might be slightly better.